### DISCUSSION TITLE
Questions about ESP32 S3 GPIOS

### DISCUSSION THREAD

**Discussion by Basti149** [arendst/Tasmota#22439] [Q&A]
Hello I wanted to buy the ESP32 S3 soon now I have a few questions about the GPIOs.
Can I use the IO39, IO40, IO41 and IO42 as normal inputs/outputs in Tasmota if I don't want to have a JTag (because in the overview I found on the internet they are marked as OK and yellow), so theoretically possible or?
The same applies to IO19 and 20 which are actually reserved for the USB, could I also use them as outputs e.g. for relay output? Maybe one or the other has a generic overview of the GPIOS for me. 
Many thanks in advance

Translated with DeepL.com (free version)

**Comment by Jason2866** [comment]
> Can I use the IO39, IO40, IO41 and IO42 as normal inputs/outputs in Tasmota

yes

> The same applies to IO19 and 20

Fully useable too

https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwi_nOzLksiJAxUHywIHHYbEGLMQFnoECAoQAQ&url=https%3A%2F%2Fwww.espressif.com%2Fsites%2Fdefault%2Ffiles%2Fdocumentation%2Fesp32-s3_datasheet_en.pdf&usg=AOvVaw195FKizqlh3gLvko2Cokxr&opi=89978449

**Reply by Basti149** [reply]
Thank you, it is important to me that I can use the 39, 40, 41, 42 as normal pins. 

The question is whether you can use 19 and 20 in Tasmota as normal GPIOS and set them under Generic or not.

**Comment by Jason2866** [comment]
Yes, but keep in mind that special GPIOs have often changing states at restart / power up
